app/models/patient.rb

```diff
@@ -118,6 +118,19 @@ def stage_changes(attributes)
     update!(pending_changes:) if pending_changes.any?
   end
 
+  def with_pending_changes
+    return self if pending_changes.blank?
+
+    dup.tap do |patient|
+      patient.clear_changes_information
+      pending_changes.each do |attr, value|
+        if patient.respond_to?("#{attr}=")
+          patient.public_send("#{attr}=", value)
+        end
+      end
```

Comment:
One for later perhaps, but I wonder if we should then clear the `pending_changes` field for this object?

It would allow us to do `patient.with_pending_changes.save!` in one go when we want to apply the pending changes.

I really like this approach though, works well!

Reply:
One issue with that is that this returns a `.dup` so the `id` field is blank; we couldn't call `.save!` on it as it will create a new record.

However, yeah that would be nice.

Reply:
Oh right, I didn't realise `dup` cleared the `id`.